addons/hr_expense/tests/test_expenses.py

```diff
@@ -296,3 +297,56 @@ def test_reconcile_payment(self):
         move = self.env['account.payment'].browse(action['res_id']).move_id
         move.button_cancel()
         self.assertEqual(sheet.state, 'cancel', 'Sheet state must be cancel when the payment linked to that sheet is canceled')
+
+    def test_print_expense_check(self):
+        """
+        Test the check content when printing a check
+        that comes from an expense
+        """
+        sheet = self.env['hr.expense.sheet'].create({
+            'company_id': self.env.company.id,
+            'employee_id': self.expense_employee.id,
+            'name': 'test sheet',
+            'expense_line_ids': [
+                (0, 0, {
+                    'name': 'expense_1',
+                    'date': '2016-01-01',
+                    'product_id': self.product_a.id,
+                    'unit_amount': 10.0,
+                    'employee_id': self.expense_employee.id,
+                }),
+                (0, 0, {
+                    'name': 'expense_2',
+                    'date': '2016-01-01',
+                    'product_id': self.product_a.id,
+                    'unit_amount': 1.0,
+                    'employee_id': self.expense_employee.id,
+                }),
+            ],
+        })
+
+        #actions
+        sheet.action_submit_sheet()
+        sheet.approve_expense_sheets()
+        sheet.action_sheet_move_create()
+        action_data = sheet.action_register_payment()
+        payment_method = self.env.company.bank_journal_ids.outbound_payment_method_ids.filtered(lambda m: m.code == 'check_printing')
+        with Form(self.env['account.payment.register'].with_context(action_data['context'])) as wiz_form:
+            wiz_form.payment_method_id = payment_method
+        wizard = wiz_form.save()
+        action = wizard.action_create_payments()
+        self.assertEqual(sheet.state, 'done', 'all account.move.line linked to expenses must be reconciled after payment')
+
+        payment = self.env['account.payment'].browse(action['res_id'])
+        pages = payment._check_get_pages()
+        stub_lines = pages[0]['stub_lines'][:1]
+        self.assertTrue(stub_lines)
+        move = self.env[action_data['context']['active_model']].browse(action_data['context']['active_ids'])
+        self.assertDictEqual(stub_lines[0], {
+            'due_date': '',
+            'number': ' - '.join([move.name, sheet.account_move_id.ref] if move.ref else [move.name]),
```

Comment:
```suggestion
            'number': ' - '.join([move.name, move.ref] if move.ref else [move.name]),
```
The value used does not match the `if` statement :wink: